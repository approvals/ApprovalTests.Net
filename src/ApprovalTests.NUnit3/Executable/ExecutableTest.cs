[TestFixture]
[UseReporter(typeof(QuietReporter))]
public class ExecutableTest
{
    static List<string> RunExecutableApproval()
    {
        var output = new List<string>();

        try
        {
            NamerFactory.AdditionalInformation = "Inner";
            Approvals.VerifyWithCallback("Sam", s => output.Add(s));
        }
        catch (Exception)
        {
        }

        return output;
    }

    [Test]
    public void TestExecutableFailure()
    {
        using (Approvals.SetFrontLoadedReporter(ReportWithoutFrontLoading.INSTANCE))
        {
            Approvals.VerifyAll(RunExecutableApproval(), "Increased feedback on");
        }
    }

    [Test]
    public void TestExecutableFailureWithPreviousApproval()
    {
        using (Approvals.SetFrontLoadedReporter(ReportWithoutFrontLoading.INSTANCE))
        {
            Approvals.VerifyAll(RunExecutableApproval(), "Increased feedback on");
        }
    }

    [Test]
    public void TestExecutableSuccess()
    {
        using (Approvals.SetFrontLoadedReporter(ReportWithoutFrontLoading.INSTANCE))
        {
            Approvals.VerifyAll(RunExecutableApproval(), "Increased feedback on");
        }
    }
}