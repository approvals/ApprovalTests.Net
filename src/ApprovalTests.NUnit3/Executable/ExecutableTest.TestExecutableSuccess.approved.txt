Increased feedback on is empty